roles/splunk/tasks/configure_firewall.yml

```diff
@@ -0,0 +1,35 @@
+---
```

Comment:
I dont have permissions to push to your branch, but here is what I'm proposing.

A bit of an explanation on what is being done here:
ufw:
1. If `ufw` was not installed on the system, and the task is installing, it seems to not have the port 22 allowd, so we need to add that as well.
2. Enableing `ufw` with the service module, does not activate the firewall rules, so we are enabling it when we add port 22
3. When we add the splunk ports,, we trigger `reloaded` with the `ufw` module.

firewalld:
1. With `firewalld` we do need to enable and start the service, and port 22 is there by default.
2. After we add the service file, firewalld needs to be reloaded, otherwise we cannot enable the new service. We cannot use the handler to do it, because it will only run at the end of the play, so we need to do it manually before enabling the new service file
3. We need to enable the service for it to activate.

```
---
- name: Ensure {{ firewall_service }}  package is installed
  ansible.builtin.package:
    name: "{{ firewall_service }}"
    state: present
  become: true

- name: Configure firewalld for Splunk
  when: firewall_service == "firewalld"
  block:
    - name: Ensure {{ firewall_service }} is Started and Enabled
      ansible.builtin.systemd:
        name: "{{ firewall_service }}"
        state: started
        enabled: true
      become: true

    - name: Add splunk firewalld service
      ansible.builtin.template:
        src: firewalld_service.xml.j2
        dest: /etc/firewalld/services/splunk.xml
        backup: true
        mode: 0644
        owner: root
        group: root
      become: true
      register: firewalld

    - name: reload firewalld
      command: firewall-cmd --reload
      become: true
      when: firewalld.changed

    - name: Activate splunk firewalld service
      ansible.posix.firewalld:
        service: splunk
        permanent: yes
        state: enabled
        immediate: true
      notify: reload firewalld
      become: true

- name: Configure UFW for Splunk
  when: firewall_service == "ufw"
  block:
    - name: Ensure SSH is enabled
      community.general.ufw:
        port: "22"
        proto: "tcp"
        rule: "allow"
        state: "enabled"
      become: true

    - name: Add splunk port to UFW
      community.general.ufw:
        port: "{{ item.number }}"
        proto: "{{ item.protocol }}"
        rule: allow
        state: "reloaded"
      become: true
      loop: "{{ splunk_firewall_ports }}"
      when: firewall_service == "ufw"
```

Reply:
You can also remove the `notify ufw` handler from `main.yml`, since it gets reloaded in the module itself.

Reply:
I added you as a contributor for future work, but I'll copypasta this for now, as well as add in @nockstarr s changes

Reply:
Done in 1acc13b0aa4e7bb55809fb2ed801ffca86b3ba3f